File: twoSum/twoSum.php
<?php
/**
 * Created by PhpStorm
 * User : Hhx
 * Date : 2019/5/7
 * Time : 15:20
 */
class Solution {

    /**
     * @param Integer[] $nums
     * @param Integer $target
     * @return Integer[]
     */
    function twoSum($nums, $target) {
        $flipNums = array_flip($nums);
        foreach($nums as $key => $value){
            if (array_key_exists($target-$value, $flipNums) && $key != $flipNums[$target-$value]) {
                return [$key, $flipNums[$target-$value]];
            }
        }
        return [];

        // $datas=array();
        // if(count($nums) >0){
        //     $num =count($nums);
        //     for($i=0;$i< $num;$i++){
        //         for($j=$i+1;$j<$num;$j++){
        //                 if($nums[$i]+$nums[$j] == $target){
        //                     $datas=[$i,$j];
        //                     break 2;
        //                 }
        //          }
        //     }
        // }
        // return $datas;

        // $num = count($nums);
        // for ($i=0; $i<$num; $i++) {
        //     $tmp = $target - $nums[$i];
        //     $newnums = $nums;
        //     unset($newnums[$i]);
        //     $res = array_search($tmp, $newnums);
        //     if ($res !== false) {
        //         return array($i,$res);
        //         break;
        //     }
        // }

        // $flipNums = array_flip($nums);
        // foreach($nums as $key => $value ){
        //     if (array_key_exists($target-$value, $flipNums) && $key != $flipNums[$target-$value]) {
        //         return [$key, $flipNums[$target-$value]];
        //     }
        // }
        // return [];
    }
}

File: twoSum/twoSum.go
package main

import "fmt"

//题目描述：给定一个整数数组 nums 和一个目标值 target，请你在该数组中找出和为目标值的那 两个 整数，并返回他们的数组下标。
//你可以假设每种输入只会对应一个答案。但是，你不能重复利用这个数组中同样的元素。
func twoSum(nums []int, target int) []int {
	var flag int
	datas :=make(map[int]int,10)
	for i:=0; i<len(nums);i++ {
		flag =target - nums[i]
		if value,ok :=datas[flag] ;ok{
			return []int{value,i}
		}else{
			datas[nums[i]] =i
		}
	}
	return []int{0}

}

func main()  {
	target := 9
	nums := []int{1,2,7,11,15}

	fmt.Print(twoSum(nums,target))
	//[1,2]


}